torch/_functorch/aot_autograd.py

```diff
@@ -866,14 +866,17 @@ class AOTConfig:
 def maybe_to_fresh_input(idx, t, meta):
     if not isinstance(t, Tensor):
         return t
-    if meta.input_info[idx].mutates_data:
-        # Make sure the primal we pass to autograd.grad()
-        # sees the tensor before the mutation
-        return t.clone()
-    if meta.input_info[idx].mutates_metadata:
-        # Make sure the primal we pass to autograd.grad()
-        # sees the tensor before the metadata mutation
-        return t.view(t.shape)
+    if idx in meta.mutated_inp_indices:
+        # We only need to bother cloning mutated inputs that participate in autograd.
```

Comment:
nit: We only need to clone mutated inputs for which we need the pre-mutation gradient.